internal/broker/broker.go

```diff
@@ -110,11 +110,16 @@ func New(cfg Config, args ...Option) (b *Broker, err error) {
 	}
 
 	if cfg.CachePath == "" {
-		return &Broker{}, errors.New("cache path must be provided")
+		err = errors.Join(err, errors.New("cache path is required and was not provided"))
 	}
-
-	if cfg.IssuerURL == "" || cfg.ClientID == "" {
-		return &Broker{}, errors.New("issuer and client ID must be provided")
+	if cfg.IssuerURL == "" {
```

Comment:
Maybe we should move the template validation here so that we have the whole validation at the same place? (it’s repeated code though), but at least, we can decide which fields are templated and which are not. I don’t any issue that way for a home dir or suffix containing `<`

Reply:
IMHO, it wouldn't make sense. The broker does not know the configuration file, so it would be weird to have it check for template values in the `broker.New` function, no?

Reply:
good point, (but right now, the parsing is done in `dbusservice` which doesn't make more sense TBH :p

I think with the double restriction with the `&&`  we can leave it being applied to all fields then, but I expect we will split that in a separate package at some point and validate required fields there too.

Reply:
> but right now, the parsing is done in dbusservice

Well, this package used to be called `brokerservice` at the time I added this parsing, so it used to make sense :joy:. But I agree that this should probably be moved to its own package later down the line, especially if the config file ends up becoming bigger and more complex.